Assets/GothicVR/Scripts/Manager/GvrSceneManager.cs

```diff
@@ -26,80 +32,146 @@ public class GvrSceneManager : SingletonBehaviour<GvrSceneManager>
         /// </summary>
         public void LoadStartupScenes()
         {
-            generalScene = SceneManager.LoadScene(generalSceneName, new LoadSceneParameters(LoadSceneMode.Additive));
-            SceneManager.MoveGameObjectToScene(interactionManager, generalScene);
-
-            LoadWorld("world.zen", "FP_PICKRICE_SWAMP_02");
+            LoadWorld("oldmine.zen", "ENTRANCE_SURFACE_OLDMINE");
+            // PxCs.Interface.PxVm.CallFunction(GameData.I.VmGothicPtr, "STARTUP_SUB_OLDCAMP"); FP_GUARD_A_OC_179
         }
 
-        public void LoadWorld(string worldName, string startVob)
+        public async void LoadWorld(string worldName, string startVob)
         {
+            await ShowLoadingScene(worldName);
             newWorldName = worldName;
             startVobAfterLoading = startVob;
-            var newWorldScene = SceneManager.LoadScene(worldName, new LoadSceneParameters(LoadSceneMode.Additive));
+            var asyncLoad = SceneManager.LoadSceneAsync(worldName, new LoadSceneParameters(LoadSceneMode.Additive));
+            Scene newWorldScene;
 
             // Remove previous scene.
             // TODO - it might be, that we need to wait for old map to be removed before loading new one. Let's see...
             if (GameData.I.WorldScene.HasValue)
             {
+                // Try to delete everything in old scene, so we can load everything again from scratch
+                var objects = GameData.I.WorldScene.Value.GetRootGameObjects();
+                for (int i = 0; i < objects.Length; i++)
+                {
+                    GameObject.Destroy(objects[i]);
+                }
                 SceneManager.UnloadSceneAsync(GameData.I.WorldScene.Value);
             }
 
-            GameData.I.WorldScene = newWorldScene;
+            asyncLoad.completed += async (asyncOperation) =>
+            {
+                newWorldScene = SceneManager.GetSceneByName(newWorldName);
```

Comment:
I think you get the scene from the completed function as parameter. Please check it. You don't need to do a search again afaik.

Reply:
removed the async loading